### PR TITLE
feat: operator implement merkle root ffi

### LINKED ISSUE #186: feat (operator): implement merkle root ffi
# TODO:

- [x] **In Rust:** Deserialize bytes from the batch.
- [x] **In Rust:** Calculate merkle root of the batch.
- [x] **In Rust:** Compare with the given merkle root.
- [ ] **In Go:** Implement binding.

### PR DESCRIPTION
**Pull request by NicolasRampoldi**
# Description
- This PR adds the Merkle Root verifier FFI.
- To test, run: 
  - Rust test: `make test_merkle_tree_rust_ffi`.
  - Finally, run make `test_merkle_tree_go_bindings_macos` or make `test_merkle_tree_go_bindings_linux` depending on your OS to test the go bindings.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -268,3 +268,26 @@ generate_sp1_fibonacci_proof:
...
 	@echo "Fibonacci proof and ELF generated in task_sender/test_examples/sp1 folder"
+
+__MERKLE_TREE_FFI__: ##
+build_merkle_tree_macos:
+	@cd operator/merkle_tree/lib && cargo build --release
+	@cp operator/merkle_tree/lib/target/release/libmerkle_tree.dylib operator/merkle_tree/lib/libmerkle_tree.dylib
+	@cp operator/merkle_tree/lib/target/release/libmerkle_tree.a operator/merkle_tree/lib/libmerkle_tree.a
+
+build_merkle_tree_linux:
+	@cd operator/sp1/lib && cargo build --release
```
Comment (taturosati) [inline Makefile]: shouldn't this be merkle_tree
```suggestion
	@cd operator/merkle_tree/lib && cargo build --release
```

File: operator/merkle_tree/lib/src/types/mod.rs
Comment (taturosati) [inline operator/merkle_tree/lib/src/types/mod.rs:1]: Couldn't this be imported from batcher?

**Review by taturosati — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (NicolasRampoldi) [reply Makefile]: Solved in [9be72b4](https://github.com/yetanotherco/aligned_layer/pull/189/commits/9be72b4be3c5920a040e2e3e899864bb798c5ee0).

File: operator/merkle_tree/lib/src/types/mod.rs
Comment (NicolasRampoldi) [reply operator/merkle_tree/lib/src/types/mod.rs:1]: Solved in [9301abb](https://github.com/yetanotherco/aligned_layer/pull/189/commits/9301abbdf9220494b2be3f0869f741e4dd4831d9).

**Review by taturosati — APPROVED** [review]
lgtm

**Review by jotabulacios — APPROVED** [review]
lgtm

**Pull request merged**